core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java

```diff
@@ -143,9 +146,11 @@ private static CloseableHttpClient init() {
                         .build());
         cm.setMaxTotal(DEFAULT_MAX_TOTAL_CONNECTIONS);
         cm.setDefaultMaxPerRoute(DEFAULT_MAX_CONNECTIONS_PER_ROUTE);
-        return HttpClients.custom()
-                .setConnectionManager(cm)
-                .useSystemProperties()
-                .build();
+        HttpClientBuilder builder = HttpClients.custom()
+                .setConnectionManager(cm);
+        if (proxy != null) {
+            builder.setProxy(proxy);
+        }
+        return builder.build();
```

Comment:
I think that if a proxy is set with our custom property/variable, then that would override the system properties. If it was not set, then the system properties' settings would still be used. I'm not sure what would happen if `useSytemProperties` and `setProxy` are used together. I hope it would work as I described, but if not, I think we can add `useSystemProperties` in an else clause on line 153.

Also, we have used the debug logging properties for the http client in the past. I'm not sure if the absence of `useSystemProperties` would prevent that.

What do you think?

Reply:
Yes, this makes sense to me 👍
(and I checked and setProxy takes precedence over useSystemProperties so we are good there)